provider/abstract_override.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'api/object'
+
+module Provider
+  # A Hash that stores overrides to the list of Api::Resource in api.yaml
```

Comment:
We need to keep the type safety here. So I'm not a fan of a generic name/value string/whatever hash. I believe it is worth to explicitly define the properties we deal with and _validate them all_. So for example if you have a new property that has to be an array, let's check the type and the array items.

Reply:
Actually, this comment lingered from a previous attempt where I was simply using a hash. This approach does do validation